### PR TITLE
Allowing value property to overridden label

### PR DESCRIPTION
**Pull request by sanghun89**
this is to avoid having items structure like this
```
{
    [ COPY.SOME_COPY ]:  { /* stuff */ }
}
```
to not have a handler based on the cop but an actual value you can base off by. Allowing:
```
{
    [ COPY.SOME_COPY ]:  {
        value: 'stuff'
    }
}
```

### CODE REVIEW

File: src/components/misc/PillToggle.jsx [hunk 1]
```diff
@@ -18,13 +19,14 @@ class PillToggle extends React.Component {
...
-		return keys.map((value, i) => {
-			const item = items[ value ];
+		return keys.map((key, i) => {
+			const item = items[ key ];
 			const { active, notifications } = item;
 			const itemClasses = classnames({
 				active,
 				item: true
 			});
+      const value = item.value || key;
```
Comment (sethen) [inline src/components/misc/PillToggle.jsx]: Why is this necessary?

Comment (sethen) [inline src/components/misc/PillToggle.jsx]: Pretty sure this needs to be tabbed over to line up with the rest of the const values.

**Comment by sethen** [comment]
Yeah, this is fine -- I agree that you should be able to have a value or a key.  Please update this pull request with the appropriate docs.

File: src/components/misc/PillToggle.jsx (hunk 1 above)
Comment (sanghun89) [inline src/components/misc/PillToggle.jsx]: wooops

File: README.md [hunk 2]
```diff
@@ -591,6 +591,7 @@ The following describes all of the miscellaneous components available for use in
 | `item`               | `obj`         | Object of items to render the key of which serves as the value of the item                     |
 | `item.active`        | `boolean`     | If item is active it will disable onclick events                                               |
 | `item.notifications` | `array`       | Takes count in first index and type in second index                                            |
+| `item.value`         | `any`         | Overrides the item's label as the designated value                                             |
```
Comment (sethen) [inline README.md]: Mm, the type is probably `string|number` here.

File: README.md [hunk 3]
```diff
@@ -617,11 +618,13 @@ class MyComponent extends React.Component {
 	render() {
 		const items = {
 			A: {
-				active: true
+				active: true,
+				value: 'foo'
```
Comment (sethen) [inline README.md]: You should keep one without the `value` key just to demonstrate both ways are acceptable.

**Comment by sethen** [comment]
Cool, squash it.
